docs/troubleshooting.md

```diff
@@ -17,6 +17,46 @@ Script analysis is provided by the [PSScriptAnalyzer] project on GitHub. If the
 message starts with `[PSScriptAnalyzer]` or if you are getting faulty script diagnostics
 (red and green squiggly lines under PowerShell in scripts) please [open an issue there].
 
+## Double-click isn't selecting the whole variable
+
+Visual Studio Code provides a default set of word separators, that is,
+characters that split words and so affect double-click selections. The editor's
+defaults include both `-` and `$`. In [v2021.5.1] we starting providing a
+default value for PowerShell files that excludes these two symbols. The
+intention of this change was to increase predictability, as double-clicking
+PowerShell symbols would now select the same portion that the extension
+highlights (with the exception of scoped variables due to the continued
+inclusion of `:` as a word separator) as well as align with collected user
+feedback.
+
+We excluded `-` because unlike programming languages that use `CamelCase` or
+`snake_case`, PowerShell uses a `Verb-Noun` style where dashes are part of many
+symbol names (like underscores in other languages). So by excluding it we teach
+Visual Studio Code to treat `Verb-Noun` as one symbol/word, which matches what
+the extension semantically highlights when the cursor is placed within it.
+
+We originally excluded `$` because PowerShell uses it as a prefix for variable
+substition, and many users were already excluding it. This change proved
+controversial and so after reviewing even more user feedback after the public
+rollout of this change, we've decided to compromise among varying user desires
+by continuing to exclude `-` but revert the exclusion of `$`.
+
+You can reset the behavior for PowerShell files to Visual Studio Code defaults
+with the following code added to `settings.json`:
+
+```json
+"[powershell]": {
+    "editor.wordSeparators": "`~!@#$%^&*()-=+[{]}\\|;:'\",.<>/?"
+}
+```
+
+Similarly you can edit this setting to exclude `$` just by removing it from the
+string, and if you do so, double-clicking `$MyVariable` will then select the
+whole symbol, matching what the extension semantically highlights, instead of
+just `MyVariable`. Keeping `-` will cause double-clicking `Verb-Noun` to only
+select the part of the symbol on the side of the `-` that you clicked. These
+settings are personal preferences.
```

Comment:
> These settings are personal preferences

As it is I read this as possibly saying "these settings are *our* personal preferences".

I'd just emphasise that everyone has a different opinion about these settings early on.
Also I suspect many users won't differentiate between the extension default and "Visual Studio Code's default" so we should word that carefully.

```suggestion
Different users have a variety of different preferences around these word highlighting settings
and [you can easily configure your own in Visual Studio Code's settings](https://stackoverflow.com/questions/31632351/visual-studio-code-customizing-word-separators).

To set the word separator behavior to separate words in PowerShell on `-` and `$`,
add the following entry to the Visual Studio Code's `settings.json`:

```json
"[powershell]": {
    "editor.wordSeparators": "`~!@#$%^&*()-=+[{]}\\|;:'\",.<>/?"
}
```

This will cause `-` and `$` to register as word boundaries,
meaning for example that double-clicking on `$MyVariable` will not select the `$`
and on the `G` in `Get-Process` will only select `Get` rather than the verb *and* noun.

You can also add or remove other characters to this string to add or remove them as word separators.
For example, removing `$` in the string will cause the `$` to be selected in `$MyVariable`
while still keeping the `Get` selection in `Get-Process`.
```